pkg/cache/tas_flavor_snapshot.go

```diff
@@ -244,6 +245,66 @@ func (s *TASFlavorSnapshot) removeTASUsage(domainID utiltas.TopologyDomainID, us
 	s.leaves[domainID].tasUsage.Sub(usage)
 }
 
+func (s *TASFlavorSnapshot) freeCapacityPerDomain() map[string]resources.Requests {
+	freeCapacityPerDomain := make(map[string]resources.Requests)
```

Comment:
```suggestion
	freeCapacityPerDomain := make(map[string]resources.Requests, len(s.leaves))
```

It would be good to pre-alloc space.